endpoints/permission/permission_endpoint.ts

```diff
@@ -0,0 +1,127 @@
+import Joi from "joi";
+import { Knex } from "knex";
+import { RedisClientType } from "redis";
+import winston from "winston";
+
+import { IBaseEndpoint } from "../../common/base_endpoint";
+import JoiValidator from "../../common/joi_validator";
+import RequestError from "../../common/RequestError";
+import PermissionModel, { TPermissions } from "../../db/models/permission";
+import UserModel, { TUser } from "../../db/models/user";
+import * as type from "./types";
+
+type CallEndpointReturnType = Record<string, never> | TPermissions | {success: true};
+
+export default class PermissionEndpoint implements IBaseEndpoint {
+  validator: JoiValidator = new JoiValidator();
+  allowNames: Array<string> = ["view", "grant", "rescind"];
+  permissionModel: PermissionModel;
+  userModel: UserModel;
+
+  constructor(public db: Knex, public redisClient: RedisClientType, public logger: winston.Logger) {
+    this.permissionModel = new PermissionModel(this.db, this.logger);
+    this.userModel = new UserModel(this.db, this.logger);
+  }
+
+  // <<< View <<<
+  async view(args: type.ViewArgs, user: TUser): Promise<TPermissions | Record<string, never>> {
```

Comment:
плохой тип возвратки
```suggestion
  async view(args: type.ViewArgs, user: TUser): Promise<TPermissions> {
```